apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
+
+Many of the values returned by embedding applications for messages of the graph module include a **subgraph**.
+
+A subgraph is a data structure which contains elements of the graph, **vertices**, and **edges** which is a precomputed lookup set of the relationships between elements of the graph.
+Consumers can pass this object to the graph module package's standard library to benefit from implementations of common operations such as getting entities linked to a particular entity.
+
+### <Hidden>Subgraph </Hidden>Structure
+
+A **subgraph** is a JSON object with the following fields, each described in detail below:
+
+- `roots`
+- `vertices`
+- `edges`
+- `depths`
+
+#### <Hidden>Subgraph Structure</Hidden> Vertices
+
+##### <Hidden>Subgraph Structure Vertices</Hidden> Vertex
+
+A `Vertex` is a JSON object which satisfies the following schema:
 
 ```json
 {
-  "$id": "https://blockprotocol.org/types/modules/graph/entity",
-  "title": "Entity",
   "type": "object",
   "properties": {
-    "entityId": {
-      "type": "string"
+    "kind": {
+      "oneOf": ["entity"]
     },
-    "entityTypeId": {
-      "type": "string"
-    },
-    "properties": {
-      "$ref": "https://example.com/path-to-the-entity-schema"
+    "inner": {
+      "$comment": "the entity",
+      "type": "object"
     }
   },
-  "required": ["entityId"]
+  "required": ["kind", "inner"]
 }
 ```
 
-##### Entity types
+> **Note**: The `kind` field is used to allow for future expansion of the subgraph format to include other kinds of vertices, such as ontology types.
+
+##### <Hidden>Subgraph Structure Vertices</Hidden> The `Vertices` object
 
-Embedding applications SHOULD also provide – via the `entityTypes` message – an **entity type** for each entity they provide the block, including the block entity. Where provided, an **entity type**:
+The `vertices` field of a **subgraph** is a JSON object where:
 
-- MUST be represented by an object, which:
-  - MUST contain an `entityTypeId` identifying the type, which MUST be a string
-  - MUST contain a `schema`, a JSON Schema object defining the constraints on the data in the entity’s `properties` field, which
-    - MUST be of `type: "object"`
-    - SHOULD use our JSON Schema extensions where appropriate
-    - MAY otherwise contain any valid JSON Schema
+- the keys are `entityId`s
+- the values are a JSON object where:
+  - the keys are `revisionId`s - a unique identifier for an instance of a specific **entity** within a subgraph, and MUST be a string. This is a forward-facing field which currently doesn't serve a functional purpose and as such can be any string.
+  - the values are `Vertex` objects
 
-i.e. entity types MUST conform to the following JSON schema:
+> **Note**: The `revisionId` will be used to identify different revisions of the same element within a single subgraph, this could be multiple versions of the same ontology type, or multiple revisions of an entity in a datastore that supports versioning
```

Comment:
```suggestion
> **Note**: The `revisionId` will be used to identify different revisions of the same element within a single subgraph. This could be multiple versions of the same ontology type, or multiple revisions of an entity in an application that supports versioning.
```